libraries/ms-drivers/src/x86/pca9539r_gpio_expander.c

```diff
@@ -58,9 +62,12 @@ static void prv_init_store(void) {
     (FreeUnpackedFunc)mx_pca9539r_store__free_unpacked,
     (UpdateStoreFunc)update_store,
   };
-  s_store.n_state = NUM_PCA9539R_GPIO_PINS;
-  s_store.state = malloc(NUM_PCA9539R_GPIO_PINS * sizeof(protobuf_c_boolean));
-  store_register(MX_STORE_TYPE__PCA9539R, funcs, &s_store, NULL);
+  for (int i = 0; i < MAX_I2C_ADDRESSES; ++i) {
+    s_stores[i] = (MxPca9539rStore)MX_PCA9539R_STORE__INIT;
+    s_stores[i].n_state = NUM_PCA9539R_GPIO_PINS;
+    s_stores[i].state = malloc(NUM_PCA9539R_GPIO_PINS * sizeof(protobuf_c_boolean));
+    store_register(MX_STORE_TYPE__PCA9539R, funcs, &s_stores[i], (void *)(intptr_t)i);
```

Comment:
can we change this to only register stores for initialized  pca9539rs? the i2c address is passed to pca9539r_gpio_init so this shouldn't be a problem.

Reply:
I asked you about that before and you said to just initialize all of them. I can change that though